test/common/schemaform/helpers.unit.spec.js

```diff
@@ -741,6 +741,33 @@ describe('Schemaform helpers:', () => {
 
       expect(output.address).to.be.undefined;
     });
+    it('should remove empty objects', () => {
+      const formConfig = {
+        chapters: {
+          chapter1: {
+            pages: {
+              page1: {}
+            }
+          }
+        }
+      };
+      const formData = {
+        page1: {
+          data: {
+            someField: {
```

Comment:
There will always be a data object underneath each page's object which has the form data. But inside that it's partly determined by how you filled out the form whether there are empty objects or undefined fields in spots. 

There's a change in this PR to replace empty objects with undefined before sending to the server, so it shouldn't matter in practice, though we could update ObjectField to clean up empty objects, too.